src/HealthChecks.OpenIdConnectServer/DependencyInjection/IdSvrHealthCheckBuilderExtensions.cs

```diff
@@ -42,5 +42,47 @@ public static class IdSvrHealthCheckBuilderExtensions
                 tags,
                 timeout));
         }
+        /// <summary>
+        /// Add a health check for Identity Server.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+        /// <param name="uriProvider">Factory for providing the uri of the Identity Server to check.</param>
+        /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'idsvr' will be used for the name.</param>
+        /// <param name="failureStatus"></param>
+        /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+        /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+        /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+        /// <returns>The specified <paramref name="builder"/>.</returns>
+        public static IHealthChecksBuilder AddIdentityServer(
+            this IHealthChecksBuilder builder,
+            Func<IServiceProvider, Uri> uriProvider,
+            string name = null,
+            HealthStatus? failureStatus = null,
+            IEnumerable<string> tags = null,
+            TimeSpan? timeout = null)
+        {
+            builder.Services.AddHttpClient();
```

Comment:
`AddHttpClient` has overload with service provider. I would you it to configure http client, get rid of `CreateIdentityServerHttpClient` method and keep `builder.Add` call the same as above.